cmd/docker/main.go

```diff
@@ -130,16 +136,30 @@ func handleFlags() bool {
 	}
 	extraconfig.Decode(src, &vchConfig)
 
-	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
-		log.SetLevel(log.DebugLevel)
-	}
-
 	*cli.portLayerAddr = fmt.Sprintf("%s:%d", *cli.portLayerAddr, *cli.portLayerPort)
 	cli.proto = "tcp"
 
 	return true
 }
 
+func initLogging() error {
+	logcfg := viclog.NewLoggingConfig()
+	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
+		logcfg.Level = log.DebugLevel
+	}
+
+	if vchConfig.Diagnostics.SysLogConfig != nil {
+		logcfg.Syslog = &syslog.SyslogConfig{
+			Network:   vchConfig.Diagnostics.SysLogConfig.Network,
+			RAddr:     vchConfig.Diagnostics.SysLogConfig.RAddr,
+			Formatter: syslog.RFC3164,
+			Priority:  srslog.LOG_INFO | srslog.LOG_DAEMON,
+		}
+	}
+
+	return viclog.Init(logcfg)
```

Comment:
@hmahmood does the hook block the log call from returning or is it async. If the former, can we push the log messages to a queue for dispatch to the syslog - I'd rather not pin our performance to that of the network & syslogd system.
Also - default TCP timeout is 10min in the case of no response, which is way too long for this scenario - I  don't know whether Go applies a different timeout value.